### ISSUE TITLE
Download queue (at least for RoyalRoad)

### ISSUE THREAD

**Issue by ka1izer** [LagradOst/QuickNovel#266]
I have a LOT of novels from royalroad.com in my library, and when I try to refresh them all, I will quickly get banned (for a time) by royalroad, since it seems the current implementation basically tries to refresh all the novels at the same time. I would think that a download queue would be the best solution, alternatively a much lower rate limit (if there is any).
My workaround for now has been to refresh right before I go to sleep, so I at least won't be affected by the ban, but it seems to be relatively random which novels actually get refreshed (since it eventually fails, always).

**Comment by LagradOst** [comment]
RoyalRoad already have a rate limit and queue (mutex) at least in the latest update, have you updated to the latest because it it working fine for me?

**Comment by ka1izer** [comment]
Yes, I had the latest version. Like I said, I had a LOT of RoyalRoad novels... Just switched phones, and have a much smaller library on the new one, so no problem for now, but...

**Comment by ka1izer** [comment]
Actually, could the problem be that I also read (a lot) directly on RoyalRoad? Maybe the rate limit is JUST within limits, and with active use on RR at the same time it's too much?
